[Python] Fix xbmcgui.Window methods to get the actual size of a Window

romanvm opened the pull request:
<!--- Provide a general summary of your change in the Title above -->

## Description
This PR fixes `xbmcgui.Window.getWidth()` and `xbmcgui.Window.getHeight()` methods so that they return the actual pixel size of a Window instance instead of the current display resolution.
Display resolution can now be obtained using `xbmcgui.getScreenWidth()` and `xbmcgui.getScreenHeight()` free functions.
Also this PR adds deprecation information to `Window.getResolution()` and `Window.setCoordinateResolution()` methods because currently they are practically useless.

## Motivation and Context
xbmcgui Controls are placed using pixel coordinates from the top left corner. However, currently there is no way to determine the the actual pixel size of the visible area (pixel coordinate grid) of a Window instance. Existing `getWidth` and `getHeight` methods return the current display resolution which is not the same as Window's coordinate grid resolution. The fixed methods will help to correctly place Controls on the screen, especially if the Controls are placed in a Window with existing ID, e.g. over a fullscreen video, and display resolution can still be obtained using the new free functions.

## How Has This Been Tested?
I compiled the code in Vusial Studio 2015 on Windows 10. Then I built a Windows installer and installed Kodi from it. After that I ran a simple testing addon and checked if those fixed methods and new functions actually work using an interactive Python debugger.

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [x] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

Commit:
Change Window.getWidth/getHeight to return real Window size

xhaggi commented:
A window should normally return it's actually width and height not the display resolution. If that's the case we should change the implementation of `GetWidth` and `GetHeight` and do not introduce new methods. Not sure if there is another way to get the current display resolution, if not that should be addressed too.

romanvm commented:
I agree with you but since this my very first PR to the Kodi core code, I didn't feel right to change the existing code. But yes, logically, a `Window` instance should return its coordinate grid resolution and the display resolution, if needed, should be returned by free functions, because it is a property of Kodi itself.

Also methods `xbmcgui::Window::getResolution ()` and `xbmcgui::Window::setCoordinateResolution (...)` are basically useless because they operate with some arcane numeric constants that are meaningless from addon developer's perspective. For example, on my Win 10 machine with 1680x1050 display `getResolution()` returns `16` or `RES_DESKTOP`: https://github.com/xbmc/xbmc/blob/master/xbmc/guilib/Resolution.h#L33.

DaveTBlake commented on xbmc/interfaces/legacy/Window.cpp:
```diff
@@ -319,7 +319,7 @@ namespace XBMCAddon
       case CGUIControl::GUICONTROL_SLIDER:
         pControl = new ControlSlider();
-        break;			
+        break;          
```
Remove trailing blanks please

Commit:
Add xbmcgui.getScreenHeight/getScreenWidth functions

Commit:
Add deprecation notice for Window.getResolution/setCoordinateResolution methods

Those methods operate with some arcane numeric constants that bear no useful information or functionality. They seem to be remnants from Xbox times with its fixed set of resolutions and are useless with modern videochips/displays.

romanvm commented:
I have updated this PR. Please see the new title and description.

xhaggi commented:
@romanvm thank you. Some minor things to do before we are ready to merge.

Please fixup/squash commit 9206e878295f8832944c1dd801483feb7662ae46 into 4fdd5bec4590efdbb522eecb80e693baafe6ce49 and c46e7ffca5f65d8f1581c659da465233c3ea3716 into 266f75c7540757338ec05250cb005ee7b01bba51.
We don't need the history for that.

romanvm commented:
Done

xhaggi commented:
Thanks looks good.

xhaggi commented:
@phil65 or @ronie any objections?

MartijnKaijser commented on xbmc/interfaces/legacy/ModuleXbmcgui.h:
```diff
@@ -83,6 +83,34 @@ namespace XBMCAddon
+    /// Returns the height of this screen.
+    ///
+    /// @return                       Screen height
+    ///
```
Shouldn't the `@python_v18` be added here as wel?

romanvm replied:
Yes, I forgot about that

romanvm commented:
@MartijnKaijser done
